app/src/main/java/org/simple/clinic/storage/Timestamps.kt

```diff
@@ -0,0 +1,42 @@
+package org.simple.clinic.storage
+
+import androidx.room.Embedded
+import org.simple.clinic.util.UtcClock
+import org.threeten.bp.Instant
+
+/**
+ * There are three timestamps that are always present in every single synced model:
+ *
+ * - `createdAt`: When the database record was created.
+ * - `updatedAt`: When the database record was updated.
+ * - `deletedAt`: When the database record was deleted.
+ *
+ * All timestamps are in UTC.
+ *
+ * This class is meant to encapsulate these timestamps as an [Embedded] model in the database
+ * entities.
+ **/
+data class Timestamps(
+    val createdAt: Instant,
+    val updatedAt: Instant,
+    val deletedAt: Instant?
+) {
+
+  companion object {
+
+    fun now(clock: UtcClock): Timestamps {
+      val instant = Instant.now(clock)
+
+      return Timestamps(createdAt = instant, updatedAt = instant, deletedAt = null)
+    }
+  }
+
+  fun updated(clock: UtcClock): Timestamps {
```

Comment:
We should be using present tense for these methods.
Like `update` and `delete`.
Also, since we are not using these yet, we could have added these methods later.

Reply:
I will rename the methods to update the tense.

Regd. whether it was necessary to add the methods now, you're right that we could have added them later. I intentionally added them now for two reasons:

- We know for a fact that we will need them because blood pressure readings can be both edited and deleted, and blood sugars will have the same features
- I wanted to give an indication to the reviewer that we can encapsulate the implementation details of updating a timestamp for editing or deleting.